### ISSUE TITLE
[Bug]: Caching - ClassificationStore KeyConfig - Pimcore 10.2.5

### ISSUE THREAD

**Issue by awroblewski** [pimcore/pimcore#11144]
### Expected behavior

Hello,

This is about creating/modifiying Classification store Keys programatically. There is no backend involvement here.

Step1 - create key
I create a classification store key using constructor \Pimcore\Model\DataObject\Classificationstore\KeyConfig, setting neccessary fields subsequently saving the KeyConfig using the ->save() method.

Step2 - modify key
Retrieving the field using KeyConfig::getByName($name, $storeId), modify the keys again saving using ->save() method.

Step - use key
Retrieve the field using KeyConfig::getByName($name, $storeId), expect to see updated data from Step 2.

### Actual behavior

Step1 - create key
I create a classification store key using constructor \Pimcore\Model\DataObject\Classificationstore\KeyConfig, setting neccessary fields subsequently saving the KeyConfig using the ->save() method -> all OK

Step2 - modify key
Retrieving the field using KeyConfig::getByName($name, $storeId), modify the keys again saving using ->save() method.

Step - use key
Retrieving the field using KeyConfig::getByName($name, $storeId), currently I see old outdated date from Step 1 as long as I do not clear the pimcore data cache using console p:c:c

### Steps to reproduce

Step 1 - create key
this is about creating classification store keys programatically. When creating a key using the \Pimcore\Model\DataObject\Classificationstore\KeyConfig constructor setting the fields subsequently saving the data using the save() method, the data is being written to the database and the correspondig cache item, if exists, is being invalidated using the calculated cache key

```
vendor/pimcore/pimcore/models/DataObject/Classificationstore/KeyConfig.php
(Line 296)            $cacheKey = 'cs_keyconfig_' . $this->getId();
(Line 297)            Cache::remove($cacheKey);
```

Step 2 - modify key
Next I want to modify keydata. First I retrieve the key by name using static KeyConfig::getByName($name, $storeId) from the KeyConfig class. This tries to retrieve the keyConfig from the cache (in my env it is redis) using the cache key

```
vendor/pimcore/pimcore/models/DataObject/Classificationstore/KeyConfig.php
(Line 155)  $cacheKey = 'cs_keyconfig_' . $storeId . '_' . md5($name);
(Line 164)  Cache::load($cacheKey);

```
This will result in a cache_miss and so the keyConfig is retreived from the database subsequently updating the cache using the key from Line 155. Now with the keyConfig I  modify the fields and again saving using the ->save() method which again invalidates the cache using the key from Line 296.

Step3 - use key
When I again want to get the KeyConfiguration by name, I get the old data from the first save action but not the updated data from Step2. The reason for this is, that the method keyConfig::getByName retrieves the KeyConfiguration using the cache key from Line 155 which was stored in Step 2, however, this key is never invalidated on save action however the cache key from Line 296 is invalidated. This leaves the old data in the cache when using KeyConfig::getByName method. Unless i either clear the Pimcore data cache using console p:c:c or invalidate the key in line 155 manually , I'll always get the old data from the first KeyConfig::getByName action in Step 2 which updates the cache using the cache key in Line 155, however this cache key is never invalidated on subsequent save actions.

To cut a long story short. If I do not miss something here I would suggest to invalidate the cache key from line 155 along with the cache key from line 296 during the KeyConfig save action.

Something like this
```
vendor/pimcore/pimcore/models/DataObject/Classificationstore/KeyConfig.php
(Line 296)            $cacheKey = 'cs_keyconfig_' . $this->getId();
(Line 297)            Cache::remove($cacheKey);

(Line xxx)            $cacheKey = 'cs_keyconfig_' . $this->getStoreId(). '_' . md5($this->getName());
(Line xxx)            Cache::remove($cacheKey);

```

This would ensure, the next call to KeyConfig::getByName would be a cache_miss subsequently retrieving the updated data from the database and updating the cache with the up to data data.

I hope this makes sense!

Andreas

**Comment by brusch** [comment]
It is possible to turn off the cache by calling `KeyConfig::setCacheEnabled(false)`, so maybe this is the right approach in your case. But you're right, it probably also makes sense to invalidate the runtime cache by calling `unset(self::$cache[$id])` . 
Could you please provide a PR for that?

**Comment by awroblewski** [comment]
Hallo,

thanks for your reply.

I do disabled the Runtime Cache using KeyConfig::setCacheEnabled(false). I think this has more todo with the redis caching invalidation. Currently I do the invalidation using $cacheKey

`$cacheKey = 'cs_keyconfig_' . $this->getStoreId(). '_' . md5($this->getName());`

in  my bundle implementation and it works fine, however, I think it is better to have the cachekey using the Key name to be invalidated along with the cachekey using the Key Id on KeyConfig save.

I have created a PR for this.

Andreas

**Comment by fashxp** [comment]
fixed via #11254
